### ISSUE TITLE
Rewrite GUI in thread safe way

### ISSUE THREAD

**Issue by unkaktus** [nogoegst/onionize#18]
`
Pango:ERROR:/build/pango1.0-hEi7pY/pango1.0-1.36.8/./pango/pango-layout.c:3916:pango_layout_check_lines: assertion failed: (!layout->log_attrs)
`
GUI is complicated as hell. What am I doing wrong??

**Comment by unkaktus** [comment]
This also seems to be related:
`(onionize:2659): Gdk-CRITICAL **: gdk_window_get_width: assertion 'GDK_IS_WINDOW (window)' failed`. 
Perhaps removing widgets is not a good idea? Sometimes it doesn't work...

**Comment by unkaktus** [comment]
It likely because I adjust widgets in another thread.

**Comment by unkaktus** [comment]
https://github.com/gotk3/gotk3-examples/blob/master/gtk-examples/goroutines/goroutines.go:
```
	// Native GTK is not thread safe, and thus, gotk3's GTK bindings may not
	// be used from other goroutines.  Instead, glib.IdleAdd() must be used
	// to add a function to run in the GTK main loop when it is in an idle
        // state.
```

**Comment by unkaktus** [comment]
Fixed in fc0e1cb5d2d0f1b5327ce191db183fd979801e83
